sdks/java/core/src/main/java/org/apache/beam/sdk/sd/BeamRowType.java

```diff
@@ -15,23 +15,23 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.apache.beam.dsls.sql.schema;
+package org.apache.beam.sdk.sd;
 
 import com.google.auto.value.AutoValue;
 import java.io.Serializable;
 import java.util.List;
 
 /**
- * Field type information in {@link BeamSqlRow}.
+ * Field type information in {@link BeamRow}.
  *
  */
 @AutoValue
-public abstract class BeamSqlRowType implements Serializable {
+public abstract class BeamRowType implements Serializable {
   public abstract List<String> getFieldsName();
   public abstract List<Integer> getFieldsType();
 
-  public static BeamSqlRowType create(List<String> fieldNames, List<Integer> fieldTypes) {
-    return new AutoValue_BeamSqlRowType(fieldNames, fieldTypes);
+  public static BeamRowType create(List<String> fieldNames, List<Integer> fieldTypes) {
```

Comment:
That would help, but it's still messy to have the relationship fall to simply having the same index in a parallel list. I'll comment on the bug, as this is somewhat orthogonal to this PR.